cg/cli/archive.py

```diff
@@ -0,0 +1,68 @@
+import click
+from click.core import ParameterSource
+
+from cg.meta.archive.archive import SpringArchiveAPI
+from cg.models.cg_config import CGConfig
+
+DEFAULT_SPRING_ARCHIVE_COUNT = 200
+
+
+@click.group()
+@click.pass_obj
+def archive():
+    """Archive utilities."""
+    pass
+
+
+@archive.command("archive-spring-files")
+@click.option(
+    "-l",
+    "--limit",
+    help="Give a limit to the amount of files to archive.",
+    default=DEFAULT_SPRING_ARCHIVE_COUNT,
+    show_default=True,
+)
+@click.option(
+    "--archive-all",
+    help="Use in order to archive all non-archived-files",
+    is_flag=True,
+    default=False,
+    show_default=True,
+)
+@click.pass_obj
+def archive_spring_files(context: CGConfig, limit: int | None, archive_all: bool):
+    """Archives non-archived spring files.
+    Raises:
+        click.Abort if both a limit to the number of spring files to archive and archive_all is specified.
+    """
+
+    if (
+        click.get_current_context().get_parameter_source("limit") == ParameterSource.COMMANDLINE
+        and limit
+        and archive_all
```

Comment:
What does the first condition do here? is `if limit and archive_all:` not enough?

Reply:
I think unless the user sets `limit = None`, the default value will be there and screw things up. So I added that to make sure that the value was set by the caller, but I will add some tests to make sure that it actually does what I intended.